File: codes/funfactorial.cpp
#include<iostream>  
  
long long factorial(int n){  
    if (n <= 1) {  
        return 1;  
    }  
    return n * factorial(n - 1);  
}  
  
int main(){  
    int n;  
    std::cout <<"请输入一个整数: ";  
    std::cin >> n;  
    if (n < 0) {  
        std::cout <<"请输入一个非负整数。" << std::endl;  
        return 1;  
    }  
    std::cout << n <<"的阶乘是: " << factorial(n) << std::endl;  
    return 0;  
}

File: codes/funfactorial.java
public class FactorialCalculator {  
  
    public static void main(String[] args) {  
        Scanner scanner = new Scanner(System.in);  
  
        System.out.print("请输入一个非负整数: ");  
        int n = scanner.nextInt();  
  
        if (n < 0) {  
            System.out.println("输入错误！请输入一个非负整数。");  
        } else {  
            long factorial = calculateFactorial(n);  
            System.out.println(n + "的阶乘是: " + factorial);  
        }  
  
        scanner.close();  
    }  
  
    public static long calculateFactorial(int n) {  
        long result = 1;  
        for (int i = 1; i <= n; i++) {  
            result *= i;  
        }  
        return result;  
    }  
}